airbyte-integrations/connectors/source-freshdesk/source_freshdesk/api.py

```diff
@@ -266,6 +279,61 @@ def list(self, fields: Sequence[str] = None) -> Iterator[dict]:
         params = {"include": "description"}
         yield from self.read(partial(self._api_get, url="tickets"), params=params)
 
+    # This block extends TicketsAPI Stream to overcome '300 page' server error.
+    # Since the TicketsAPI Stream list has a 300 page pagination limit, after 300 pages, update the parameters with
+    # query using 'updated_since' = last_record, if there is more data remaining.
+    @staticmethod
+    def get_tickets(
+        result_return_limit: int, getter: Callable, params: Mapping[str, Any] = None, ticket_paginate_limit: int = 300
+    ) -> Iterator:
+        """Read using getter"""
+        params = params or {}
+        # the maximum page allowed to pull during pagination.
+        # ticket_paginate_limit 300
+        # Start page
+        page = 1
+        # Initial request parameters
+        params = {**params, "per_page": result_return_limit}
+
+        while True:
+            params["page"] = page
+            batch = list(getter(params=params))
+            yield from batch
+
+            if len(batch) < result_return_limit:
+                return iter(())
+
+            # get last_record from latest batch, pos. 0, because of DESC order of records
+            last_record = batch[0]["updated_at"]
+
+            # checkpoint & switch the pagination
+            if page == ticket_paginate_limit:
+                page = 0  # reset page counter
+                last_record = pendulum.parse(last_record).add(seconds=1)
```

Comment:
add a comment why we need to add 1 sec, and what happens if we do not do it.

Reply:
Done